networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandler.java

```diff
@@ -57,10 +74,12 @@ public void onIncomingMessage(
     if (storageClient.getStore() != null) {
       SafeFuture<Void> future = SafeFuture.COMPLETE;
       if (!peer.wantToMakeRequest() || !peer.wantToReceiveObjects(callback, message.size())) {
-        peer.disconnectCleanly(DisconnectReason.RATE_LIMITING).ifExceptionGetsHereRaiseABug();
```

Comment:
im assuming this is redundant and so was removed, but gotta check..

Reply:
oh yes. Forgot about that - it was very early in my investigations.  The `peer.wantToMakeRequest` and `peer.wantToReceiveObjects` calls automatically handle the rate limiting disconnect. This was the only place we were doing it manually.